### PR TITLE
Clarify role of `--system` flag in README

### LINKED ISSUE #3951: `--system` flag no longer working as documented
I have been using `uv` in virtual environments managed by `pyenv virtualenv`. The `--system` flag worked great with `uv` 0.1.45 but does not work with 0.2.5, and the docs still suggest that it should still work:
> For convenience, `uv pip install --system` will install into the system Python environment, as an approximate shorthand for, e.g., `uv pip install --python=$(which python3)`. 

I understand that, currently, there are two work-arounds:
- `uv pip ... --python=$(which python)`
- `python -m uv ...`

But `--system` was more convenient, and it wasn't clear to me that it was intentional for it to have stopped working as previously.

Successful usage of `--system` flag with 0.1.45:
```sh
$ venv_name=uv-0.1.x
$ pyenv uninstall --force ${venv_name}
$ pyenv virtualenv ${venv_name}
$ pyenv shell ${venv_name}
$ pip install uv==0.1.45

[details omitted]

$ uv --version
uv 0.1.45 (a33a05e2d 2024-05-20)
$ pyenv version
uv-0.1.x (set by PYENV_VERSION environment variable)
$ python -c 'import sys; print(f"{(sys.prefix==sys.base_prefix)=}")'
(sys.prefix==sys.base_prefix)=False
$ uv pip install --system --upgrade pip setuptools wheel
Resolved 3 packages in 299ms
Installed 3 packages in 34ms
 - pip==22.3.1
 + pip==24.0
 - setuptools==65.5.0
 + setuptools==70.0.0
 + wheel==0.43.0
$ pyenv shell --unset
```

Unsuccessful usage of `--system` flag with 0.2.5:
```sh
$ venv_name=uv-0.2.x
$ pyenv uninstall --force ${venv_name}
$ pyenv virtualenv ${venv_name}
$ pyenv shell ${venv_name}
$ pip install uv

[details omitted]

$ uv --version
uv 0.2.5 (47db418ba 2024-05-28)
$ pyenv version
uv-0.2.x (set by PYENV_VERSION environment variable)
$ python -c 'import sys; print(f"{(sys.prefix==sys.base_prefix)=}")'
(sys.prefix==sys.base_prefix)=False
$ uv pip install --system --upgrade pip setuptools wheel --verbose
DEBUG Searching for Python interpreter in search path
DEBUG Found CPython 3.11.3 at `/Users/dpoznik/.pyenv/versions/uv-0.2.x/bin/python3` (search path)
DEBUG Ignoring Python interpreter at `/Users/dpoznik/.pyenv/versions/uv-0.2.x/bin/python3`: system interpreter required
DEBUG Found CPython 3.11.3 at `/Users/dpoznik/.pyenv/versions/uv-0.2.x/bin/python` (search path)
DEBUG Ignoring Python interpreter at `/Users/dpoznik/.pyenv/versions/uv-0.2.x/bin/python`: system interpreter required
DEBUG Found CPython 3.11.3 at `/Users/dpoznik/.pyenv/shims/python3` (search path)
DEBUG Ignoring Python interpreter at `/Users/dpoznik/.pyenv/versions/uv-0.2.x/bin/python3`: system interpreter required
DEBUG Found CPython 3.11.3 at `/Users/dpoznik/.pyenv/shims/python` (search path)
DEBUG Ignoring Python interpreter at `/Users/dpoznik/.pyenv/versions/uv-0.2.x/bin/python`: system interpreter required
DEBUG Found CPython 3.12.3 at `/usr/local/bin/python3` (search path)
DEBUG Using Python 3.12.3 environment at /usr/local/opt/python@3.12/bin/python3.12
error: The interpreter at /usr/local/opt/python@3.12/Frameworks/Python.framework/Versions/3.12 is externally managed, and indicates the following:

[additional details omitted]

$ uv pip install --python=$(which python) --upgrade pip setuptools wheel
Resolved 3 packages in 562ms
Uninstalled 2 packages in 654ms
Installed 3 packages in 29ms
 - pip==22.3.1
 + pip==24.0
 - setuptools==65.5.0
 + setuptools==70.0.0
 + wheel==0.43.0
$ python -m uv pip install --upgrade pip setuptools wheel
Resolved 3 packages in 1.08s
Audited 3 packages in 0.22ms
$ pyenv shell --unset
```
Thanks for your help with this issue and for developing this wonderful tool!

### PR DESCRIPTION
**Pull request by zanieb**
For uv>=0.2.0 behavior

Closes #3951

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,18 +176,29 @@ search for a Python interpreter matching that version in the following order:
...
+Although we generally recommend using virtual environments for dependency management,
+`--system` is appropriate in continuous integration and containerized environments.
+
+The `--system` flag is also used to opt in to mutating system environments. For example, the
+the `--python` argument can be used to request a Python version e.g. `--python 3.12` and uv will
+search for an interpreter that meets the request. If uv finds a system interpreter, e.g., `/usr/lib/python3.12`,
+then the `--system` flag is required to allow modification of this non-virtual Python environment.
+Without the `--system` flag, uv will ignore any interpreters that are not in virtual environments.
+Conversely, when the `--system` flag is provided, uv will ignore any interpreters that are
+in virtual environments.
```
Comment (dpoznik) [inline README.md:201]:
```suggestion
The `--system` flag is also used to opt in to mutating system environments. For example, the
the `--python` argument can be used to request a Python version (e.g., `--python 3.12`), and uv will
search for an interpreter that meets the request. If uv finds a system interpreter (e.g., `/usr/lib/python3.12`),
then the `--system` flag is required to allow modification of this non-virtual Python environment.
Without the `--system` flag, uv will ignore any interpreters that are not in virtual environments.
Conversely, when the `--system` flag is provided, uv will ignore any interpreters that *are*
in virtual environments.
```

**Review by dpoznik — APPROVED** [review]
This reads well and clarifies a few points. Thanks!
Some minor typographical suggestions below; feel free to ignore.

File: README.md [hunk 2]
```diff
@@ -176,18 +176,29 @@ search for a Python interpreter matching that version in the following order:
...
+a directory that is **not** a [PEP 405 compliant](https://peps.python.org/pep-0405/#specification)
+virtual environment, it will be ignored.
+
+uv can also install into arbitrary, even non-virtual environments, with the `--python` argument
+provided to `uv pip sync` or `uv pip install`. For example, `uv pip install --python=/path/to/python`
+will install into the environment linked to the `/path/to/python` interpreter.
+
+For convenience, `uv pip install --system` will install into the system Python environment.
+Using `--system` is roughly equivalent to `uv pip install --python=$(which python)`,
+but note it will skip `python` executables on the `PATH` that execute in a virtual environment.
```
Comment (charliermarsh) [inline README.md:191]: Nit: "but note that"

Comment (charliermarsh) [reply README.md:191]: Could "executables on the `PATH` that execute in a virtual environment." be something like... "executables that are linked to virtual environments"?

File: README.md [hunk 3]
```diff
@@ -176,18 +176,29 @@ search for a Python interpreter matching that version in the following order:
...
-
-For convenience, `uv pip install --system` will install into the system Python environment, as an
-approximate shorthand for, e.g., `uv pip install --python=$(which python3)`. Though we generally
-recommend the use of virtual environments for dependency management, `--system` is intended to
-enable the use of uv in continuous integration and containerized environments.
+`/path/to/venv`, regardless of where uv is installed. Note that if `VIRTUAL_ENV` is set to
+a directory that is **not** a [PEP 405 compliant](https://peps.python.org/pep-0405/#specification)
+virtual environment, it will be ignored.
+
+uv can also install into arbitrary, even non-virtual environments, with the `--python` argument
```
Comment (charliermarsh) [inline README.md:185]: This is confusing because `--system` is _required_ for this now, right?

**Review by charliermarsh — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (zanieb) [reply README.md:185]: No, if it's explicit (i.e. that path to the executable or its directory) we do not require the `--system` flag.

File: README.md (hunk 2 above)
Comment (zanieb) [reply README.md:191]:
```suggestion
but note that executables that are linked to virtual environments will be skipped.
```

**Pull request merged**